src/Pages/App/CreateAppModal/AppClone/useDevtronCloneList.tsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2024. Devtron Inc.
+ */
+import { useEffect, useMemo, useRef, useState } from 'react'
+
+import {
+    abortPreviousRequests,
+    getIsRequestAborted,
+    showError,
+    useAsync,
+    useStateFilters,
+} from '@devtron-labs/devtron-fe-common-lib'
+
+import { getJobs } from '@Components/Jobs/Service'
+import { APP_TYPE } from '@Config/constants'
+import { getAppListMin } from '@Services/service'
+
+import { getDevtronAppList } from '../service'
+import { DevtronAppCloneListProps, DevtronListResponse } from './types'
+
+export const useDevtronCloneList = ({ handleCloneAppClick, isJobView }: DevtronAppCloneListProps) => {
+    const cloneListAbortControllerRef = useRef(new AbortController())
+    const [isLoadingMore, setIsLoadingMore] = useState(false)
+    const [hasError, setHasError] = useState(false)
+    const { searchKey } = useStateFilters()
+
+    const fetchDevtronCloneList = (offset = 0): Promise<DevtronListResponse> =>
+        abortPreviousRequests(
+            () =>
+                isJobView
+                    ? getJobs(
+                          {
+                              teams: [],
+                              appStatuses: [],
+                              appNameSearch: '',
+                              offset,
+                              size: 20,
+                              sortBy: 'appNameSort',
+                              sortOrder: 'ASC',
+                              searchKey,
+                          },
+                          {
+                              signal: cloneListAbortControllerRef.current.signal,
+                          },
+                      ).then((res) => ({ type: APP_TYPE.JOB, data: res }))
+                    : getAppListMin(null, { signal: cloneListAbortControllerRef.current.signal }).then((res) => ({
+                          type: APP_TYPE.DEVTRON_APPS,
+                          data: res,
+                      })),
```

Comment:
We should create a service for this block and move the list and totalCount computation inside the serive itself
```suggestion
                isJobView
                    ? getJobs(
                          {
                              teams: [],
                              appStatuses: [],
                              appNameSearch: '',
                              offset,
                              size: 20,
                              sortBy: 'appNameSort',
                              sortOrder: 'ASC',
                              searchKey,
                          },
                          {
                              signal: cloneListAbortControllerRef.current.signal,
                          },
                      ).then((res) => ({ type: APP_TYPE.JOB, data: res }))
                    : getAppListMin(null, { signal: cloneListAbortControllerRef.current.signal }).then((res) => ({
                          type: APP_TYPE.DEVTRON_APPS,
                          data: res,
                      })),
```